codex/sales/slotqueue.nim

```diff
@@ -205,6 +215,14 @@ proc activeWorkers*(self: SlotQueue): int =
 proc contains*(self: SlotQueue, item: SlotQueueItem): bool =
   self.queue.contains(item)
 
+proc pause*(self: SlotQueue) =
+  # set unpaused flag to false -- unblocks coroutines waiting on unpaused.wait()
+  self.unpaused.clear()
+
+proc unpause*(self: SlotQueue) =
+  # set unpaused flag to true -- coroutines will block on unpaused.wait()
+  self.unpaused.fire()
+
```

Comment:
It's the other way around; `clear()` ensures that `wait()`s are blocked, `fire()` ensures that `wait()`s go through. The documentation of `AsyncEvent` is not very good on this point, but the description of `fire()` is clear: https://status-im.github.io/nim-chronos/api/chronos/asyncsync.html#fire%2CAsyncEvent

```suggestion
proc pause*(self: SlotQueue) =
  # set unpaused flag to false -- coroutines will block on unpaused.wait()
  self.unpaused.clear()

proc unpause*(self: SlotQueue) =
  # set unpaused flag to true -- unblocks coroutines waiting on unpaused.wait()
  self.unpaused.fire()

```

Reply:
Thanks for the clarification. I had forgotten to update the comments, but got it
right in the logic 😅

Reply:
I know where I got this mixed up. Look at the [documentation for the constructor for `AsyncEvent`](https://status-im.github.io/nim-chronos/api/chronos/asyncsync.html#AsyncEvent): 
> An event manages a flag that can be set to true with the fire() procedure and reset to false with the clear() procedure. _The wait() coroutine blocks until the flag is false._ (emphasis mine)

This is actually wrong, or at least I seemed to keep reading it in way that says the flag must be false to unblock coroutines.